tools/msbuild-fuzzer/Program.cs

```diff
@@ -0,0 +1,311 @@
+				AutomaticNuGetRestore = false,
+				CleanupAfterSuccessfulBuild = false,
+				CleanupOnDispose = true,
+				Root = Xamarin.Android.Build.Paths.TestOutputDirectory,
+			}) {
+				directory = Path.GetFullPath (Path.Combine (builder.Root, builder.ProjectDirectory));
+				if (Directory.Exists (directory))
+					Directory.Delete (directory, recursive: true);
+
+				application = new XamarinFormsAndroidApplicationProject ();
+				application.AndroidManifest = application.AndroidManifest.Replace ("<uses-sdk />", "<uses-sdk android:targetSdkVersion=\"28\" />");
+				application.MainActivity = application.DefaultMainActivity.Replace ("//${AFTER_ONCREATE}", "Android.Util.Log.Debug (\"FUZZER\", \"App started!\");");
+				var abis = new string [] { "armeabi-v7a", "arm64-v8a", "x86" };
+				application.SetProperty (KnownProperties.AndroidSupportedAbis, string.Join (";", abis));
+
+				if (!NuGetRestore ()) {
+					Console.WriteLine ("Initial NuGet restore failed!");
+					return;
+				}
+
+				Func<bool> [] operations = {
+					NuGetRestore,
+					Build,
+					DesignTimeBuild,
+					DesignerBuild,
+					Clean,
+					Install,
+					Uninstall,
+					Run,
+					TouchRandomFile,
+					ChangePackageName,
+					AddClass,
+					RemoveClass,
+					RenameClass,
+					AddResource,
+					RemoveResource,
+					RenameResource,
+				};
+
+				while (true) {
+					var operation = operations [random.Next (operations.Length)];
+					if (!operation ()) {
+						break;
+					}
+				}
+			}
+
+			Console.WriteLine ("Press enter to exit...");
+			Console.ReadLine ();
+		}
+
+		static bool NuGetRestore ()
+		{
+			Console.WriteLine (nameof (NuGetRestore));
+			return builder.RunTarget (application, "Restore", doNotCleanupOnUpdate: true);
+		}
+
+		static bool Build ()
+		{
+			Console.WriteLine (nameof (Build));
+			return builder.Build (application, doNotCleanupOnUpdate: true);
+		}
+
+		static bool DesignTimeBuild ()
+		{
+			Console.WriteLine (nameof (DesignTimeBuild));
+			return builder.DesignTimeBuild (application, doNotCleanupOnUpdate: true);
+		}
+
+		static readonly string [] DesignerParameters = new [] { "DesignTimeBuild=True", "AndroidUseManagedDesignTimeResourceGenerator=False" };
+
+		static bool DesignerBuild ()
+		{
+			Console.WriteLine (nameof (DesignerBuild));
+			return builder.RunTarget (application, "SetupDependenciesForDesigner", doNotCleanupOnUpdate: true, parameters: DesignerParameters);
+		}
+
+		static bool Clean ()
+		{
+			Console.WriteLine (nameof (Clean));
+			return builder.Clean (application, doNotCleanupOnUpdate: true);
+		}
+
+		static bool Install ()
+		{
+			Console.WriteLine (nameof (Install));
+			if (!builder.Install (application, doNotCleanupOnUpdate: true)) {
+				return false;
+			}
+			if (!installedPackages.Contains (application.PackageName))
+				installedPackages.Add (application.PackageName);
+			needsInstall = false;
+			return true;
+		}
+
+		static bool Uninstall ()
+		{
+			foreach (var packageName in installedPackages) {
```

Comment:
Shouldn't there also be a:

```csharp
Console.WriteLine (name (Uninstall));
```

This would be consistent with the other methods.